presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -1105,6 +1096,18 @@ protected Type visitNode(Node node, StackableAstVisitorContext<Context> context)
             throw new SemanticException(NOT_SUPPORTED, node, "not yet implemented: " + node.getClass().getName());
         }
 
+        public Type visitGroupingOperation(GroupingOperation node, StackableAstVisitorContext<Context> context)
+        {
+            if (node.getGroupingColumns().size() > MAX_NUMBER_GROUPING_ARGUMENTS) {
+                throw new SemanticException(INVALID_PROCEDURE_ARGUMENTS, node, "GROUPING supports up to 63 column arguments");
+            }
+            for (Expression columnArgument : node.getGroupingColumns()) {
+                process(columnArgument, context);
```

Comment:
I'm not sure what you're talking about. Is this in relation to the equality vs identity problem? If so, I have a solution https://github.com/prestodb/presto/pull/7712#discussion_r115062316.